test/es6/generators-syntax.js

```diff
@@ -212,6 +212,14 @@ var tests = [
             assert.throws(function () { eval("function *gf() { (a = (yield) => {}) => {}; }"); }, SyntaxError, "yield expression is disallowed within arrow function default parameter expression in nested case too", "The use of a keyword for an identifier is invalid");
         }
     },
+    {
+        name : "yield is allowed before 'in' in a for loop control but not elsewhere bug issue #5203",
+        body : function () {
+            assert.doesNotThrow(function () { eval("function* gf() {for(var a = yield in {});}"); }, "Yield is allowed before 'in' in a for loop");

```

Comment:
What is the correct interpretation of `for(var a = yield 1 in {}) {…`? I think the only way for that to parse is as `for( var a = (yield 1) in {})`, is that how it behaves with this change?

Reply:
And going further afield, it looks like v8 parses ` for(var a = yield 'a' in {} in {a: 1})` as `for(var a = (yield 'a') in ({} in {a:1}))`; does that work out here as well?

Reply:
Both of these work in that way - I'll compose some functionality tests.